service/history/shard/context_impl.go

```diff
@@ -1661,25 +1688,61 @@ func (s *ContextImpl) loadShardMetadata(ownershipChanged *bool) error {
 func (s *ContextImpl) GetReplicationStatus(cluster []string) (map[string]*historyservice.ShardReplicationStatusPerCluster, map[string]*historyservice.HandoverNamespaceInfo, error) {
 	remoteClusters := make(map[string]*historyservice.ShardReplicationStatusPerCluster)
 	handoverNamespaces := make(map[string]*historyservice.HandoverNamespaceInfo)
+	clusterInfo := s.clusterMetadata.GetAllClusterInfo()
 	s.rLock()
 	defer s.rUnlock()
 
 	if len(cluster) == 0 {
 		// remote acked info for all known remote clusters
-		for k, v := range s.remoteClusterInfos {
-			remoteClusters[k] = &historyservice.ShardReplicationStatusPerCluster{
-				AckedTaskId:             v.AckedReplicationTaskID,
-				AckedTaskVisibilityTime: timestamp.TimePtr(v.AckedReplicationTimestamp),
+		for clusterName, v := range s.remoteClusterInfos {
+			if _, ok := clusterInfo[clusterName]; !ok {
+				continue
+			}
+			var taskID int64
```

Comment:
Maybe I misread.
taskID is 0, how can `v.AckedReplicationTaskIDs[remoteShardID] < taskID` be true?